subprojects/build-init/src/main/java/org/gradle/buildinit/plugins/internal/JvmProjectInitDescriptor.java

```diff
@@ -185,7 +192,26 @@ private void addStandardDependencies(BuildScriptBuilder buildScriptBuilder) {
         if (getLanguage() == Language.SCALA) {
             String scalaVersion = libraryVersionProvider.getVersion("scala");
             String scalaLibraryVersion = libraryVersionProvider.getVersion("scala-library");
-            buildScriptBuilder.implementationDependency("Use Scala " + scalaVersion + " in our library project", "org.scala-lang:scala-library:" + scalaLibraryVersion);
+            String scalaCoordinates = constraintsDefined ? "org.scala-lang:scala-library" : "org.scala-lang:scala-library:" + scalaLibraryVersion;
+            buildScriptBuilder.implementationDependency("Use Scala " + scalaVersion + " in our library project", scalaCoordinates);
+        }
+    }
+
+    private void addDependencyConstraints(BuildScriptBuilder buildScriptBuilder) {
+        String commonsTextVersion = libraryVersionProvider.getVersion("commons-text");
+        buildScriptBuilder.implementationDependencyConstraint("Define dependency versions as constraints", "org.apache.commons:commons-text:" + commonsTextVersion);
+
+        if (getLanguage() == Language.GROOVY) {
+            buildScriptBuilder.implementationDependencyConstraint("Use the latest Groovy version for building this library", "org.codehaus.groovy:groovy-all:" + libraryVersionProvider.getVersion("groovy"));
```

Comment:
:x: The `"Use the latest Groovy/Scala/Kotlin version..."` comments should not be added two times. I think it is fine to keep it on the dependency and remove it here from the constraints. (same for KOTLIN and SCALA below)